one-ui/ui/src/components/entities/mapping/mapping-card.tsx

```diff
@@ -103,6 +110,142 @@ const MappingCard: React.FC<Props> = (props) => {
         <span style={{fontSize: '16px'}}>Are you sure you want to delete this?</span>
         </Modal>;
 
+
+    const getSourceData = async (index) => {
+
+        let database = props.data[index].sourceDatabase || 'data-hub-STAGING';
```

Comment:
This seems like implementation details that the GUI shouldn't have to worry about. Could the GUI instead say - Hey middle tier, get me a source document for this step?

Reply:
@rjrudin - This is how it works in the current QS as well. We pass the source database name and Source Query to the middle-tier via the API '/search/sjsSearch' to get the list of URIs and then supply one of those URIs to the endpoint '/search/doc' to get the required Doc.
If we want to send only the step number then we will have to redesign this process but the hurdle here is that a mapping artifact will not have a step number unless its added t a flow. Either ways should we take that up in this story itself?

Reply:
Yes we can revisit later, I may open a ticket to rework this.